### PR TITLE
Update "Proxy Configuration" tutorial

### LINKED ISSUE #1125: Update proxy tutorial
* In the intro, state that "some" Clear applications benefit from autoproxy.  Not all.  For example, mixer still requires manual proxy settings.

* "Shells and programs in a desktop session" - These proxy settings are not unique to "Desktop".  Remove "desktop" from the header.

* For the .bashrc settings, add a note about applying it globally to /etc/profile.d/proxy.conf.

* Add proxy settings for autospec/DNF

* The Docker section shows how to apply the proxys and verify.  Should do the same for the others.

### PR DESCRIPTION
**Pull request by bktan8**
Closes #1125

Signed-off-by: Bun K Tan <bun.k.tan@intel.com>

### CODE REVIEW

File: source/guides/maintenance/proxy.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  For detailed instructions on installing |CL| on a bare metal system, visit
+  the :ref:`bare metal installation guide <bare-metal-install-desktop>`.
+
+General proxy settings for many applications
+============================================
+
+Proxy settings:
+
+  .. code-block:: none
```
[the remark this answers is not in the document]
Comment (bktan8) [reply source/guides/maintenance/proxy.rst]: done.

File: source/guides/maintenance/proxy.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+When working behind a corporate proxy server, one typically has to configure
+proxy settings for applications to reach the Internet.  |CL-ATTR| has 
+implemented an :ref:`autoproxy` feature to try and eliminate manual 
+configurations as much as possible.  However, there are still some applications 
+that cannot take full advantage of the :ref:`autoproxy` feature due to their 
+own ways of configuring proxy settings.  This guide shows you how to configure 
+proxy settings for some of the known applications manually.  
+
+feature.
```
[the remark this answers is not in the document]
Comment (bktan8) [reply source/guides/maintenance/proxy.rst]: done.

**Comment by mvincerx** [comment]
LGTM.

**Comment by bktan8** [comment]
@puneetse  - any comments on this doc?

**Comment by bktan8** [comment]
@mvincerx - we're good to merge.

**Comment by mvincerx** [comment]
LGTM

**Pull request merged**